### PR TITLE
Address consistent period printing on Julia 1.5

### PR DESCRIPTION
**Pull request by omus**
Should now finally be fixed with: https://github.com/JuliaLang/julia/pull/34387

### CODE REVIEW

**Comment by omus** [comment]
I'll need to check all of the various printing to ensure this package is consistent

**Comment by omus** [comment]
Last part of this that I'll leave until later: https://github.com/JuliaTime/TimeZones.jl/issues/248

File: src/io.jl [hunk 1]
```diff
@@ -101,7 +101,7 @@ function Base.show(io::IO, t::Transition)
 end
 
 function Base.show(io::IO, zdt::ZonedDateTime)
-    if get(io, :compact, false) || get(io, :compact_el, false)
+    if VERSION <= v"1.5.0-DEV.163" && get(io, :compact_el, false)
```
Comment (fchorney) [inline src/io.jl]: don't we still want to check for `:compact` here?

**Comment by omus** [comment]
A quick outline of how things print now:

Julia 1.5-DEV with PR:
```julia
julia> using TimeZones

julia> d = ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
2020-02-04T11:20:00-06:00

julia> show(d)
ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
julia> show([d])
[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show([d d])
[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show(IOContext(stdout, :compact=>true), [d, d])
[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg"), ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show(IOContext(stdout, :compact=>false), [d, d])
[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg"), ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> [d, d][1]
2020-02-04T11:20:00-06:00

julia> fill(d, 2)
2-element Array{ZonedDateTime,1}:
 2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00

julia> fill(d, 2, 2)
2×2 Array{ZonedDateTime,2}:
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00

julia> print(d)
2020-02-04T11:20:00-06:00
julia> string(d)
"2020-02-04T11:20:00-06:00"

julia> repr(d)
"ZonedDateTime(2020, 2, 4, 11, 20, tz\"America/Winnipeg\")"

julia> fill(d => d, 2)
2-element Array{Pair{ZonedDateTime,ZonedDateTime},1}:
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") => ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") => ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")

julia> fill(d => d, 2, 3)
2×3 Array{Pair{ZonedDateTime,ZonedDateTime},2}:
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")  …  ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")     ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
```

Julia 1.0 with PR
```julia
julia> using TimeZones

julia> d = ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
2020-02-04T11:20:00-06:00

julia> show(d)
ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
julia> show([d])
ZonedDateTime[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show([d d])
ZonedDateTime[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show(IOContext(stdout, :compact=>true), [d, d])
ZonedDateTime[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg"), ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> show(IOContext(stdout, :compact=>false), [d, d])
ZonedDateTime[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg"), ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> [d, d][1]
2020-02-04T11:20:00-06:00

julia> fill(d, 2)
2-element Array{ZonedDateTime,1}:
 2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:0

julia> fill(d, 2, 2)
2×2 Array{ZonedDateTime,2}:
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00

julia> print(d)
2020-02-04T11:20:00-06:00
julia> string(d)
"2020-02-04T11:20:00-06:00"

julia> repr(d)
"ZonedDateTime(2020, 2, 4, 11, 20, tz\"America/Winnipeg\")"

julia> fill(d => d, 2)
2-element Array{Pair{ZonedDateTime,ZonedDateTime},1}:
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") => ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg") => ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")

julia> fill(d => d, 2, 3)
2×3 Array{Pair{ZonedDateTime,ZonedDateTime},2}:
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")  …  ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
 ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")     ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")=>ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
```

Julia 1.0 before PR:
```julia
julia> using TimeZones

[ Info: Recompiling stale cache file /Users/omus/.julia/compiled/v1.0/TimeZones/2uWP2.ji for TimeZones [f269a46b-ccf7-5d73-abea-4c690281aa53]

julia> d = ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
2020-02-04T11:20:00-06:00

julia> show(d)
ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")
julia> show([d])
ZonedDateTime[2020-02-04T11:20:00-06:00]
julia> show([d d])
ZonedDateTime[2020-02-04T11:20:00-06:00 2020-02-04T11:20:00-06:00]
julia> show(IOContext(stdout, :compact=>true), [d, d])
ZonedDateTime[2020-02-04T11:20:00-06:00, 2020-02-04T11:20:00-06:00]
julia> show(IOContext(stdout, :compact=>false), [d, d])
ZonedDateTime[ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg"), ZonedDateTime(2020, 2, 4, 11, 20, tz"America/Winnipeg")]
julia> [d, d][1]
2020-02-04T11:20:00-06:00

julia> fill(d, 2)
2-element Array{ZonedDateTime,1}:
 2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00

julia> fill(d, 2, 2)
2×2 Array{ZonedDateTime,2}:
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00

julia> print(d)
2020-02-04T11:20:00-06:00
julia> string(d)
"2020-02-04T11:20:00-06:00"

julia> repr(d)
"ZonedDateTime(2020, 2, 4, 11, 20, tz\"America/Winnipeg\")"

julia> fill(d => d, 2)
2-element Array{Pair{ZonedDateTime,ZonedDateTime},1}:
 2020-02-04T11:20:00-06:00 => 2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00 => 2020-02-04T11:20:00-06:00

julia> fill(d => d, 2, 3)
2×3 Array{Pair{ZonedDateTime,ZonedDateTime},2}:
 2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00
 2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00  2020-02-04T11:20:00-06:00=>2020-02-04T11:20:00-06:00
```

File: src/io.jl (hunk 1 above)
Comment (omus) [reply src/io.jl]: I don't think so? See the output example: https://github.com/JuliaTime/TimeZones.jl/pull/247#issuecomment-582023513

Comment (fchorney) [reply src/io.jl]: If we're running julia <= v"1.5.0-DEV.163"
then `show(IOContext(stdout, :compact=>true), d)`  would end up printing the non-compact version here without that check that was removed no?

Comment (omus) [reply src/io.jl]: This is also true for `TimeType`s on Julia 1.5-DEV.

```julia
julia> show(IOContext(stdout, :compact=>true), Date(2000))
Date("2000-01-01")
```

I'll note that this PR is trying to mirror the 1.5 behaviour on all versions of Julia

Comment (fchorney) [reply src/io.jl]: ahh ok. that makes sense.

**Comment by omus** [comment]
Bumped project version and squashed commits

**Pull request merged**